source/views/settings/screens/licenses/index.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import * as React from 'react'
+import {View, StyleSheet} from 'react-native'
+import {LicensesList} from './list'
+import {type LicenseType} from './types'
+import {type NavigationScreenProp} from 'react-navigation'
+
+import licenseData from '../../../../../docs/licenses.json'
+
+type Props = {
+	navigation: NavigationScreenProp<*>,
+}
+
+export class LicensesView extends React.Component<Props> {
+	static navigationOptions = {
+		title: 'Open Source Licenses',
+	}
+
+	extractNameFromGithubUrl = (url: string) => {
+		if (!url) {
+			return null
+		}
+
+		const reg = /((https?:\/\/)?(www\.)?github\.com\/)?(@|#!\/)?([A-Za-z0-9_]{1,15})(\/([-a-z]{1,20}))?/iu
+		const components = reg.exec(url)
+
+		if (components && components.length > 5) {
+			return components[5]
+		}
+		return null
```

Comment:
I think ES2018's [named capture groups](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges) might be great solution here. The prior groups are apparently needed for optional matching, but if we were to just use something like `(?<username>[a-z0-9]*)`, then we could just yank out `components.groups.username`.

Reply:
I'm open to exploring implementing it with named capture groups!